deployment/amplify/staging/amplify/backend/function/echolocatorStagingProfiles/src/app.js

```diff
@@ -0,0 +1,180 @@
+  next()
+});
+
+
+app.post('/profiles', function(req, res) {
+  var region = process.env.REGION;
+  AWS.config.update({region: region});
+  var s3 = new AWS.S3();
+  var cognito = new AWS.CognitoIdentityServiceProvider();
+
+  var identityId = req.body.identityId;
+  var email = req.body.email;
+  if (!identityId) {
+    res.json({error: 'Missing Cognito user identityId in POST body'});
+    return;
+  } else {
+    // sanity-check identity ID looks like [AWS region]:[UUID]
+    var idParts = identityId.split(':')
+    if (!idParts || idParts.length !== 2 ||
+      idParts[0] !== region || !validateUuid(idParts[1])) {
+
+      console.error('Cognito identity ID sent does not match expected format');
+      res.json({error: 'Cognito identity ID does not match expected format', identityId});
+      return;
+    }
+  }
+  if (!email) {
+    res.json({error: 'Missing user email in POST body'});
+    return;
+  }
+
+  // get Cognito user pool ID from CloudFront env parameters
+  var userPool = process.env.AUTH_ECHOLOCATORSTAGINGAUTH_USERPOOLID;
+  var bucketName = process.env.STORAGE_ECHOLOCATORSTAGINGSTORAGE_BUCKETNAME;
+  console.log('Bucket name: ' + bucketName);
+
+  cognito.adminGetUser({
+    UserPoolId: userPool,
+    Username: email
+  }, function(getUserError, userData) {
+    if (getUserError && getUserError.code === 'UserNotFoundException') {
+      console.error('User with email ' + email + 'does not exist');
+      res.json({error: 'User does not exist', email: email});
+    } else if (getUserError) {
+      console.error('Failed to get user');
+      console.error(getUserError);
+      res.json({error: err});
+    } else {
+      console.log('Found user');
+      console.log(userData);
+
+      // get voucher number from user data
+      if (!userData || !userData.UserAttributes || userData.UserAttributes.length === 0) {
+        // Shouldn't happen, but check anyways
+        console.error('Returned user data is missing attributes');
+        console.error(userData);
+        res.json({error: 'Returned user data is missing attributes'});
+        return;
+      }
+
+      // Look through the name/value pairs for the voucher number
+      var voucher = null;
+      for (var i = 0; i < userData.UserAttributes.length; i++) {
+        var attr = userData.UserAttributes[i];
+        if (attr.Name === 'custom:voucher') {
+          voucher = attr.Value;
+          break;
+        }
+      }
+
+      if (!voucher) {
+        console.error('Failed to find voucher number for user account');
+        res.json({error: 'User has no voucher number in Cognito', email});
+        return;
+      }
+
+      s3.listObjectsV2({
+        Bucket: bucketName,
+        Prefix: 'public/' + voucher
+      }, function (listError, listData) {
+        if (listError) {
+          console.error('S3 query failed');
+          console.error(listError);
+          res.json({error: listError});
+        } else {
+          console.log('S3 query succeeded', listData);
```

Comment:
The logs can be viewed in the Lambda AWS interface, and I've found them useful for debugging. I assume the storage cost for that is minimal, but now the functions are working, I agree it would make sense to trim down the informational messages a bit.